### ISSUE TITLE
looking up transform from frame [base_link] to frame [map]

### ISSUE THREAD

**Issue by yani-rl-ai** [NilsDengler/sem_mapping#3]
Hi @NilsDengler  i have achieved this following step

Prerequisites for mapping:

- the map of the environment has to be loaded

- amcl or other localizer has to run

important commands:

- build the repo in release mode: catkin config --profile release -x _release --cmake-args -DCMAKE_BUILD_TYPE=Release && catkin build --profile release

run tensorflow:

- source ~/tensorflow/bin/activate && roslaunch tensorflow_object_detector object_detect.launch

However, when i run this step : 
rosrun semmapping mapping _camera_info:=YOUR/CAMERA/INFO/TOPIC

i got this error. 
[ERROR] [1675307587.926233218, 51.552000000]: Lookup would require extrapolation 0.387000000s into the past.  Requested time 51.534000000 but the earliest data is at time 51.921000000, when looking up transform from frame [base_link] to frame [map]
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_footprint_wheel at time 51.554000 according to authority unknown_publisher
         at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.6/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_footprint at time 51.554000 according to authority unknown_publisher

i do not understand why the error is related to map and base_link. maybe you have some clues about this error.

**Comment by yani-rl-ai** [comment]
hi @NilsDengler could you help me? sorry for inconvenient

**Comment by NilsDengler** [comment]
Hi @ROSNovice,
sorry for letting you wait. 
Please provide me more details about your setup:
What ROS version do you currently use? 
Do you playback your data using a rosbag or do you use a live recording setup (real robot)?
Do you get this error from the beginning or only after some time?

**Comment by yani-rl-ai** [comment]
hi @NilsDengler 
thanks for the reply
1. i use ros noetic and i also the toyota HSR user. 
2. i build the package with "catkin_make", not use your tutorial command
3. i used gazebo simulation with Toyota HSR 
4. before i run this command "rosrun semmapping mapping _camera_info:=YOUR/CAMERA/INFO/TOPIC"
i load the 2D occupancy map that i obtained from gmapping, AMCL and navigation stack package (TERMINAL 1), and then i run the object detection from Tensorflow API (TERMINAL 2 with virtual environment). since the ros noetic used python 3, i used the tensorflow 2. 
5.  i got this below error in the beginning after i run "rosrun semmapping mapping _camera_info:=YOUR/CAMERA/INFO/TOPIC" (TERMINAL 3 with virtual environment)
 [ERROR] [1675307587.926233218, 51.552000000]: Lookup would require extrapolation 0.387000000s into the past. Requested time 51.534000000 but the earliest data is at time 51.921000000, when looking up transform from frame [base_link] to frame [map]
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_footprint_wheel at time 51.554000 according to authority unknown_publisher
at line 278 in /tmp/binarydeb/ros-noetic-tf2-0.7.6/src/buffer_core.cpp
Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame base_footprint at time 51.554000 according to authority unknown_publisher

**Comment by yani-rl-ai** [comment]
hi, could you help me?

**Comment by NilsDengler** [comment]
Does this error only happen once at the beginning, or does it happen all the time?
I would say this is the corresponding line: https://github.com/NilsDengler/sem_mapping/blob/e8ad484ca74f618129d29870a7953e3f85bbc7e7/Src/semmapping/src/mapping.cpp#L416

**Comment by yani-rl-ai** [comment]
hi @NilsDengler  thank you again for answering my question. 
the error happen all the time,  i was successfully run https://github.com/Eruvae/hypermap in ROS NOETIC using HSR. i also can visualize the result in Rviz. does this package depend on the hypermap rviz plugin to show the visualization results ?

**Comment by NilsDengler** [comment]
You are right! I uploaded my version of the hypermap rviz plugin. Please check if it works now.

**Comment by yani-rl-ai** [comment]
hi @NilsDengler your hypermap rviz plugin need the hypermap_msgs, where the messages are crash with mapping_msgs version. please help me
what():  Tried to subscribe to a topic with the same name but different md5sum as a topic that was already subscribed [hypermap_msgs/HypermapMetaData/5befb9e9353d4c71ff05aaa102cd721a vs. mapping_msgs/SemanticMap/c1b150d6270bd6f62161d4550237b3a1]
